src/scores/continuous/correlation/correlation_impl.py

```diff
@@ -47,14 +47,129 @@ def pearsonr(
+        preserve_dims: Optionally specify which dimensions to preserve when
+            calculating the Spearman's rank correlation coefficient. All other dimensions will
+            be reduced. As a special case, 'all' will allow all dimensions to be
+            preserved.
+
+    Returns:
+        An xarray object with Spearman's rank correlation coefficient values.
+
+    Raises:
+        ValueError: If `preserve_dims` is set to 'all', a ValueError will be raised.
+        TypeError: If the input types are not xarray DataArrays or Datasets.
+        ValueError: If the input Datasets do not have the same data variables.
+
+    Note:
+        This function isn't set up to take weights.
+
+    Reference:
+        Spearman, C. (1904). The Proof and Measurement of Association between Two Things. The American Journal of Psychology, 15(1), 72–101.
+        https://doi.org/10.2307/1412159
+
+    See also:
+        :py:func:`scores.continuous.correlation.pearsonr`
+
+    Example:
+        >>> import xarray as xr
+        >>> import numpy as np
+        >>> from scores.continuous.correlation.correlation_impl import spearmanr
+        >>> # Create example forecast and observation DataArrays
+        >>> fcst = xr.DataArray(np.random.rand(10, 5), dims=("time", "location"))
+        >>> obs = xr.DataArray(np.random.rand(10, 5), dims=("time", "location"))
+        >>> # Calculate Spearman's rank correlation coefficient
+        >>> result = spearmanr(fcst, obs, reduce_dims="time")
+        >>> print(result)
+    """
+    reduce_dims = scores.utils.gather_dimensions(
+        fcst.dims, obs.dims, reduce_dims=reduce_dims, preserve_dims=preserve_dims
+    )
+    if preserve_dims == "all":
+        raise ValueError(
+            "The 'preserve_dims' argument cannot be set to 'all' for the Spearman's correlation coefficient."
+        )
+
+    def _spearman_calc(fcst, obs, reduce_dims):
+        """
+        Core calculation for Spearman's rank correlation coefficient.
+        """
+        # Flatten/stack the arrays along the dimensions to be reduced
+        tmp_dim = "".join(list(fcst.dims) + list(obs.dims))
+        obs_stacked = obs.stack({tmp_dim: reduce_dims})
+        fcst_stacked = fcst.stack({tmp_dim: reduce_dims})
+        # Rank
+        fcst_ranks = fcst_stacked.rank(dim=tmp_dim)
+        obs_ranks = obs_stacked.rank(dim=tmp_dim)
+
+        return pearsonr(fcst_ranks, obs_ranks, reduce_dims=tmp_dim)
+
+    if isinstance(fcst, xr.DataArray) and isinstance(obs, xr.DataArray):
+        return _spearman_calc(fcst, obs, reduce_dims)
+
+    if isinstance(fcst, xr.Dataset) and isinstance(obs, xr.Dataset):
+        # Ensure both datasets have the same variables
+        if set(fcst.data_vars) != set(obs.data_vars):
+            raise ValueError("Both datasets must contain the same variables.")
+
+        # Apply spearmanr to each variable
+        results = {var: _spearman_calc(fcst[var], obs[var], reduce_dims) for var in fcst.data_vars}
```

Comment:
Optional

```suggestion
        results = {_v: _spearman_calc(fcst[_v], obs[_v], reduce_dims) for _v in fcst.data_vars}
```

`var` isn't a reserved word, but its close to `vars` and a bit too common to be used in a generator/comprehension. Obfuscating it with `_var` or `_v` may be better.

Reply:
Updated